Library/Homebrew/formula.rb

```diff
@@ -1318,18 +1318,25 @@ def verify_download_integrity(fn)
 
   # @private
   def run_test
-    old_home = ENV["HOME"]
-    build, self.build = self.build, Tab.for_formula(self)
-    mktemp do
-      @testpath = Pathname.pwd
-      ENV["HOME"] = @testpath
-      setup_home @testpath
-      test
+    preserve_files = ARGV.keep_files?
+    begin
+      old_home = ENV["HOME"]
+      build, self.build = self.build, Tab.for_formula(self)
+      mktemp("#{name}-", { :preserve => preserve_files }) do
+        @testpath = Pathname.pwd
+        ENV["HOME"] = @testpath
+        setup_home @testpath
+        test
+      end
+    ensure
+      if preserve_files
+        ohai "Kept temporary files"
+        puts "Temporary files are saved at #{@testpath}"
+      end
```

Comment:
`ohai` gets truncated to the terminal width so it could lose the path information, and that's necessary for this to be useful. And I didn't want to do just `puts` because without the `ohai` header it tends to get lost in the build output